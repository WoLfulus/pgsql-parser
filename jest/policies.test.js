const parser = require('../src');
import { cleanTree, cleanLines } from './utils';
const dosql = require('fs').readFileSync(__dirname + '/fixtures/policies.sql').toString();

describe('policies', () => {
  it('parse and deparse', () => {
    const tree = parser.parse(dosql);
    expect(tree).toMatchSnapshot();
    const sql = parser.deparse(tree.query);
    expect(cleanLines(sql)).toMatchSnapshot();
    expect(cleanTree(parser.parse(sql))).toEqual(cleanTree(tree));
  });
});
