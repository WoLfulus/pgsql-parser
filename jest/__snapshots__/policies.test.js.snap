// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`policies parse and deparse 1`] = `
Object {
  "query": Array [
    Object {
      "RawStmt": Object {
        "stmt": Object {
          "CreatePolicyStmt": Object {
            "cmd_name": "delete",
            "permissive": true,
            "policy_name": "delete_user",
            "qual": Object {
              "A_Expr": Object {
                "kind": 0,
                "lexpr": Object {
                  "ColumnRef": Object {
                    "fields": Array [
                      Object {
                        "String": Object {
                          "str": "id",
                        },
                      },
                    ],
                    "location": 80,
                  },
                },
                "location": 83,
                "name": Array [
                  Object {
                    "String": Object {
                      "str": "=",
                    },
                  },
                ],
                "rexpr": Object {
                  "TypeCast": Object {
                    "arg": Object {
                      "FuncCall": Object {
                        "args": Array [
                          Object {
                            "A_Const": Object {
                              "location": 101,
                              "val": Object {
                                "String": Object {
                                  "str": "user.id",
                                },
                              },
                            },
                          },
                        ],
                        "funcname": Array [
                          Object {
                            "String": Object {
                              "str": "current_setting",
                            },
                          },
                        ],
                        "location": 85,
                      },
                    },
                    "location": 111,
                    "typeName": Object {
                      "TypeName": Object {
                        "location": 113,
                        "names": Array [
                          Object {
                            "String": Object {
                              "str": "uuid",
                            },
                          },
                        ],
                        "typemod": -1,
                      },
                    },
                  },
                },
              },
            },
            "roles": Array [
              Object {
                "RoleSpec": Object {
                  "location": 54,
                  "rolename": "authenticated",
                  "roletype": 0,
                },
              },
            ],
            "table": Object {
              "RangeVar": Object {
                "inh": true,
                "location": 29,
                "relname": "user",
                "relpersistence": "p",
                "schemaname": "users",
              },
            },
          },
        },
        "stmt_len": 119,
      },
    },
  ],
  "stderr": "",
}
`;

exports[`policies parse and deparse 2`] = `"CREATE POLICY \\"delete_user\\" ON \\"users\\" . \\"user\\" FOR DELETE TO \\"authenticated\\" USING ((\\"id\\") = (current_setting('user.id')::uuid));"`;
